backtracking/rat_in_maze.py

```diff
@@ -0,0 +1,167 @@
+def fill_matrix(matrix, i, j):
+    """
+    This method just fills matrix i, j with value
+    Parameters :
+        matrix(2D matrix) : matrix
+        i, j : coordinates of the matrix.
+    """
+    matrix[i][j] = 1
+
+
+def solveMaze(maze):
+    """
+    This method solves rat in maze algorithm.
+    In this problem we have n by n matrix and we have start point and end point
+    we want to go from source to distination. In this matrix 0 are block paths
+    1 are open paths we can use.
+    Parameters :
+        maze(2D matrix) : maze
+        size : size of our maze(square)
+    Returns:
+        This method returns true and array if solution found otherwise false and None
+
+    >>> maze = [[0, 1, 0, 1, 1],
+    ...         [0, 0, 0, 0, 0],
+    ...         [1, 0, 1, 0, 1],
+    ...         [0, 0, 1, 0, 0],
+    ...         [1, 0, 0, 1, 0]]
+    >>> solveMaze(maze)
+    [1, 0, 0, 0, 0]
+    [1, 1, 1, 1, 0]
+    [0, 0, 0, 1, 0]
+    [0, 0, 0, 1, 1]
+    [0, 0, 0, 0, 1]
+
+    >>> maze = [[0, 1, 0, 1, 1],
+    ...         [0, 0, 0, 0, 0],
+    ...         [0, 0, 0, 0, 1],
+    ...         [0, 0, 0, 0, 0],
+    ...         [0, 0, 0, 0, 0]]
+    >>> solveMaze(maze)
+    [1, 0, 0, 0, 0]
+    [1, 0, 0, 0, 0]
+    [1, 0, 0, 0, 0]
+    [1, 0, 0, 0, 0]
+    [1, 1, 1, 1, 1]
+
+    >>> maze = [[0, 0, 0],
+    ...         [0, 1, 0],
+    ...         [1, 0, 0]]
+    >>> solveMaze(maze)
+    [1, 1, 1]
+    [0, 0, 1]
+    [0, 0, 1]
+
+    >>> maze = [[0, 1, 0],
+    ...         [0, 1, 0],
+    ...         [1, 0, 0]]
+    >>> solveMaze(maze)
+    Solution does not exists!
+
+    >>> maze = [[0, 1],
+    ...         [1, 0]]
+    >>> solveMaze(maze)
+    Solution does not exists!
+    """
+    size = len(maze)
+    # We need to create solution object to save path.
+    solutions = [[0 for _ in range(size)] for _ in range(size)]
+    solved = runmaze(maze, 0, 0, solutions)
+
+    if solved:
+        result = "".join(str(row) + "\n" for row in solutions)
+        print(result[: len(result) - 1])  # For last \n
```

Comment:
```suggestion
        print("\n".join(str(row) for row in solutions))
```

Reply:
For testing my part of the code was more comfortable. If it is not big Problem can we leave this part as it is?